cpp/oneapi/dal/backend/primitives/reduction/reduction_rm_cw_dpc.hpp

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/reduction/functors.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+class kernel_reduction_rm_cw_naive;
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+class reduction_rm_cw_naive {
+public:
+    using inp_t = const Float*;
+    using out_t = Float*;
+    using kernel_t = kernel_reduction_rm_cw_naive<Float, BinaryOp, UnaryOp>;
+
+public:
+    reduction_rm_cw_naive(sycl::queue& q, const std::int64_t wg);
+    reduction_rm_cw_naive(sycl::queue& q);
+    sycl::event operator()(inp_t input,
+                           out_t output,
+                           const std::int64_t width,
+                           const std::int64_t stride,
+                           const std::int64_t height,
```

Comment:
We do not require `const` for parameters passed by value. The same in many other functions.

```suggestion
                           std::int64_t width,
                           std::int64_t stride,
                           std::int64_t height,
```

Reply:
Won't fix.